lib/microformats/format_parser.rb

```diff
@@ -347,10 +122,262 @@ def check_for_h_properties
+          child.is_a?(Nokogiri::XML::Element) && child.name == 'a' && !child.attribute('href').nil?
+        end
+
+        if child_a_tags_with_href.count == 1
+          node = child_a_tags_with_href.first
+
+          if format_classes(node).empty?
+            @properties['url'] = [node.attribute('href').value.strip]
+          end
+        end
+
+        if @properties['url'].nil?
+          # else if .h-x>area[href]:only-of-type:not[.h-*], then use that [href] for url
+          child_area_tags_with_href = element.children.select do |child|
+            child.is_a?(Nokogiri::XML::Element) && child.name == 'area' && !child.attribute('href').nil?
+          end
+
+          if child_area_tags_with_href.count == 1
+            node = child_area_tags_with_href.first
+
+            if format_classes(node).empty?
+              @properties['url'] = [node.attribute('href').value.strip]
+            end
+          end
+        end
+
+        child_elements = element.children.reject { |child| child.is_a?(Nokogiri::XML::Text) }
+
+        if @properties['url'].nil? && child_elements.count == 1 && format_classes(child_elements.first).empty?
+          child_element = child_elements.first
+          # else if .h-x>:only-child:not[.h-*]>a[href]:only-of-type:not[.h-*], then use that [href] for url
+          child_a_tags_with_href = child_element.children.select do |child|
+            child.is_a?(Nokogiri::XML::Element) && child.name == 'a' && !child.attribute('href').nil?
+          end
+
+          if child_a_tags_with_href.count == 1
+            node = child_a_tags_with_href.first
+
+            if format_classes(node).empty?
+              @properties['url'] = [node.attribute('href').value.strip]
+            end
+          end
+
+          if @properties['url'].nil?
+            # else if .h-x>:only-child:not[.h-*]>area[href]:only-of-type:not[.h-*], then use that [href] for url
+            child_area_tags_with_href = child_element.children.select do |child|
+              child.is_a?(Nokogiri::XML::Element) && child.name == 'area' && !child.attribute('href').nil?
+            end
+
+            if child_area_tags_with_href.count == 1
+              node = child_area_tags_with_href.first
+
+              if format_classes(node).empty?
+                @properties['url'] = [node.attribute('href').value.strip]
+              end
+            end
+          end
+        end
+      end
+
+      unless @properties['url'].nil?
+        @properties['url'] = [Microformats::AbsoluteUri.new(@properties['url'].first, base: @base).absolutize]
+      end
+    end
+
+    def prefix_from_class(class_name)
+      class_name.downcase.split('-')[0]
+    end
+
+    def property_from_class(class_name)
+      class_name.downcase.split('-')[1..-1].join('-')
+    end
+
+    def imply_dates(element)
+      ### imply date for dt-end if dt-start is defined with a date ###
+      if !@properties['end'].nil? && !@properties['start'].nil?
```

Comment:
Since this conditional wraps the entire body of this method, I _think_ it could be rewritten as a guard class:

```ruby
return unless @properties['end'] && @properties['start']
```

That might not be 100% accurate, though, so double-check that logic.